frame/reward-distribution/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+                tvl_percentage / distro_params.ideal_dapps_staking_tvl * adjustable_balance
+            };
+
+            let total_staker_balance = base_staker_balance + adjustable_staker_part;
+            let total_treasury_balance =
+                base_treasuy_balance + (adjustable_balance - adjustable_staker_part);
+
+            // Prepare imbalances
+            let (dapps_imbalance, remainder) = reward.split(dapps_balance);
+            let (stakers_imbalance, remainder) = remainder.split(total_staker_balance);
+            let (collator_imbalance, remainder) = remainder.split(collator_balance);
+            let (treasury_imbalance, remainder) = remainder.split(total_treasury_balance);
+
+            // TODO: remove this later, useful for testing atm
+            assert!(remainder.peek().is_zero());
+
+            // Payout beneficiaries
+            T::BeneficiaryPayout::treasury(treasury_imbalance);
+            T::BeneficiaryPayout::collators(collator_imbalance);
+            T::BeneficiaryPayout::dapps_staking(stakers_imbalance, dapps_imbalance);
+
+            // TODO: benchmark this code
+            100
+        }
+    }
+}
+
+// TODO (remove this)
+// Alternative to this approach would be to have some as configurable constants.
+// This approach, however, provides more flexibility since we can change them without
+// runtime upgrade.
+
+/// List of parameters used to calculate reward distribution portions for all the beneficiaries.
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct RewardDistributionParams {
+    /// Base percentage of reward that goes to treasury
+    #[codec(compact)]
+    pub base_treasury_percent: Perbill,
+    /// Base percentage of reward that goes to stakers
+    #[codec(compact)]
+    pub base_staker_percent: Perbill,
+    /// Percentage of rewards that goes to dApps
+    #[codec(compact)]
+    pub dapps_percent: Perbill,
+    /// Percentage of reward that goes to collators
+    #[codec(compact)]
+    pub collators_percent: Perbill,
+    /// Adjustable reward percentage that either goes to treasury or to stakers
+    #[codec(compact)]
+    pub adjustable_percent: Perbill,
+    /// Target dapps-staking TVL percentage at which adjustable inflation towards stakers becomes saturated
+    #[codec(compact)]
+    pub ideal_dapps_staking_tvl: Perbill,
+}
+
+impl Default for RewardDistributionParams {
+    fn default() -> Self {
+        RewardDistributionParams {
+            base_treasury_percent: Perbill::from_percent(40),
+            base_staker_percent: Perbill::from_percent(25),
+            dapps_percent: Perbill::from_percent(25),
+            collators_percent: Perbill::from_percent(10),
+            adjustable_percent: Zero::zero(),
+            ideal_dapps_staking_tvl: Zero::zero(),
+        }
+    }
+}
+
+impl RewardDistributionParams {
+    /// `true` if sum of all percentages is `one whole`, `false` otherwise.
+    fn verify(&self) -> bool {
```

Comment:
May be `is_consistent` a bit more descriptive rather than `verify`.

Reply:
Changed.